Trimble.Modus.Components/Controls/Button/TMButton.cs

```diff
@@ -0,0 +1,407 @@
+﻿
+using Microsoft.Maui.Controls.Shapes;
+
+
+namespace Trimble.Modus.Components
+{
+
+    public class TMButton : ContentView
+    {
+        private readonly Label _titleLabel;
+        private readonly Image _iconImage;
+        private readonly TapGestureRecognizer _tapGestureRecognizer;
+        private bool imageSet = false;
+        private bool isTextSet = false;
+        private Border frame;
+        private StackLayout stackLayout;
+        private bool sizeSet = false;
+        public static readonly BindableProperty TitleProperty =
+            BindableProperty.Create(nameof(Title), typeof(string), typeof(TMButton), propertyChanged: OnTitleChanged);
+
+        public static readonly BindableProperty IconSourceProperty =
+            BindableProperty.Create(nameof(IconSource), typeof(ImageSource), typeof(TMButton), propertyChanged: OnIconSourceChanged);
+
+        public static readonly BindableProperty CommandProperty =
+            BindableProperty.Create(nameof(Command), typeof(Command), typeof(TMButton), null);
+
+        public static readonly BindableProperty CommandParameterProperty =
+            BindableProperty.Create(nameof(CommandParameter), typeof(object), typeof(TMButton), null);
+
+        public static readonly BindableProperty SizeProperty =
+            BindableProperty.Create(nameof(Size), typeof(ButtonSize), typeof(TMButton), propertyChanged: OnSizeChanged);
+
+        public static readonly BindableProperty IsFloatingButtonProperty =
+            BindableProperty.Create(nameof(IsFloatingButton), typeof(bool), typeof(TMButton), false, propertyChanged: OnIsFloatingButtonChanged);
+
+        public static readonly BindableProperty IsDisabledProperty =
+            BindableProperty.Create(nameof(IsDisabled), typeof(bool), typeof(TMButton), false, propertyChanged: OnIsDisabledChanged);
+
+        public new static readonly BindableProperty BackgroundColorProperty = 
+            BindableProperty.Create(nameof(BackgroundColor), typeof(Color), typeof(TMButton), (Color)BaseComponent.colorsDictionary()["TrimbleBlue"],propertyChanged:onBackgroundColorChanged);
+
+        public static readonly BindableProperty BorderColorProperty =
+            BindableProperty.Create(nameof(BorderColor), typeof(Color), typeof(TMButton), (Color)BaseComponent.colorsDictionary()["TrimbleBlue"], propertyChanged: onBorderColorChanged);
+       
+        public static readonly BindableProperty TextColorProperty = 
+            BindableProperty.Create(nameof(TextColor), typeof(Color), typeof(TMButton), defaultValue: Colors.White, propertyChanged: onTextColorChanged);
+       
+        public static readonly BindableProperty ButtonRadiusProperty =
```

Comment:
@ElishaSamPeterPrabhu Please have this default button radius as a constant in the top of the class

Reply:
Defined as Constant